flow/src/main/scala/org/alephium/flow/core/TxUtils.scala

```diff
@@ -452,6 +452,244 @@ trait TxUtils { Self: FlowUtils =>
     }
   }
 
+  def partitionOutputs(
+      outputs: AVector[TxOutputInfo],
+      totalGasOpt: Option[GasBox]
+  )(implicit
+      groupConfig: GroupConfig
+  ): AVector[(AVector[TxOutputInfo], Option[GasBox])] = {
+    val outputsByGroup = outputs.groupBy(_.lockupScript.groupIndex(groupConfig))
+    val gasPerOutputOpt = totalGasOpt.map { gas =>
+      val totalDestinations = outputsByGroup.view.mapValues(_.length).values.sum
+      Math.round(gas.value.toFloat / totalDestinations) // primitive rounding
```

Comment:
I haven't dig the details yet, but are you sure it's fine doing a rounding here?
I remember when working on those endpoints that we can quickly run out of gas for just few atto ALPH.
If the user provide exactly the right amount of gas, with the rounding we might lose just enough to have not enough gas in the end.

Reply:
Probably the BigInteger base dividing in the `U256` implementation : 
```
gas.toU256.divUnsafe(U256.unsafe(totalDestinations))
```

Reply:
Got it, I will add a `div` method to `GasBox` as it contains all three add, sub, mul but not div yet.

Reply:
sorry for the late answer, yes that's the idea. I did that [here](https://github.com/alephium/alephium/blob/f3c69898878a298dd87aa2b5b186a73f0dcb48e2/flow/src/main/scala/org/alephium/flow/core/TxUtils.scala#L891-L896)